### PR TITLE
DEVPROD-4233 Set task ID as attribute

### PR DESCRIPTION
**Pull request by bsamek**
DEVPROD-4233

### Description

Including the name of the task in the span makes it counintuitive to search for
the span, as typically the name of a span is the name of the function that it is
in, and is therefore static. This adds the task.id as an attribute instead of in
the span name.

We may wish to announce in Slack that this change is being made, as it may break
existing queries.

### Testing

None.

### Documentation

None.

### CODE REVIEW

File: agent/agent.go
Comment (ybrill) [inline agent/agent.go:610]: There should be a task ID attribute set on the span already (`evergreen.task.id`).
Is it missing from this span?

File: agent/agent.go
Comment (ybrill) [inline agent/agent.go:609]: do we want to do the same for the [function spans](https://github.com/evergreen-ci/evergreen/blob/26c086e9bd295bf13d87add1e9f0e5da107ddc35/agent/command.go#L187)?

Comment (bsamek) [reply agent/agent.go:609]: I'm torn about this. On the one hand, it seems like the correct thing to do. On the other hand, it would make viewing a trace harder: https://ui.honeycomb.io/mongodb-4b/environments/production/datasets/evergreen-agent/result/xGe4zGJjqpi/trace/477vTF73h2v?fields[]=s_name&fields[]=s_serviceName&span=36e4a995c90acfeb. On the other hand, you can add the fields as a column.

I made the change though. Maybe I can ask in Slack before I merge?

File: agent/agent.go
Comment (bsamek) [reply agent/agent.go]: Ah, true, removed.

File: agent/agent.go
Comment (ybrill) [reply agent/agent.go:609]: Either way's good by me 😄

**Review by ybrill — APPROVED** [review]

**Review by ablack12 — COMMENTED** [review]
No need to wait on my review but does anything need to be added or updated here?
https://docs.devprod.prod.corp.mongodb.com/devprod-docs/observability/Honeycomb/honeycomb-exploring-evergreen-data

**Comment by bsamek** [comment]
> No need to wait on my review but does anything need to be added or updated here? https://docs.devprod.prod.corp.mongodb.com/devprod-docs/observability/Honeycomb/honeycomb-exploring-evergreen-data

I don't think so.

**Pull request merged**
